libraries/AP_RangeFinder/AP_RangeFinder_BLPing.h

```diff
@@ -50,17 +92,81 @@ class AP_RangeFinder_BLPing : public AP_RangeFinder_Backend_Serial
         CRC_H
     };
 
-    uint32_t last_init_ms;      // system time that sensor was last initialised
-    uint16_t distance_cm;       // latest distance
-
-    // structure holding latest message contents
+    /**
+     * @brief Structure holding the last message available with its state
+     *
+     */
     struct {
-        ParseState state;       // state of incoming message processing
+        ParserState state;      // state of incoming message processing
+        bool done;              // inform if the message is complete or not
         uint8_t payload[20];    // payload
         uint16_t payload_len;   // latest message payload length
-        uint16_t msgid;         // latest message's message id
+        uint16_t id;            // latest message's message id
         uint16_t payload_recv;  // number of message's payload bytes received so far
         uint16_t crc;           // latest message's crc
         uint16_t crc_expected;  // latest message's expected crc
     } msg;
 };
+
+/**
+ * @brief Class for Blue Robotics Ping1D sensor
+ *
+ */
+class AP_RangeFinder_BLPing : public AP_RangeFinder_Backend_Serial
+{
+    static constexpr uint16_t _sensor_rate_ms = 100; // initialise sensor at no more than 10hz
```

Comment:
I believe we should use the maximum frequency. Why did you choose 10Hz?

Reply:
There is no rate limit in the software and 10Hz looks good enough for our application, keep in mind that the original codes uses 1Hz.
Let me know if you have a better rate in mind.

Reply:
use 50ms for now please (20Hz). The high update rate is a useful feature of the device that I think should be used.

Reply:
This results in unstable answer time from the sensor, floating around 14Hz and 20Hz.
From my tests, 10Hz is a stable sample time without timeouts. 
![image](https://user-images.githubusercontent.com/1215497/74250794-e9086380-4cc9-11ea-98d9-f2272bd0675a.png)